metadata-editor/app/lib/MetadataMessageConsumer.scala

```diff
@@ -0,0 +1,34 @@
+package lib
+
+import play.api.libs.json._
+import com.gu.mediaservice.lib.aws.MessageConsumer
+import com.gu.mediaservice.model.Image
+import play.api.libs.concurrent.Execution.Implicits._
+
+import scala.concurrent.Future
+
+object MetadataMessageConsumer extends MessageConsumer(
+  Config.queueUrl, Config.awsEndpoint, Config.awsCredentials, MetadataEditorMetrics.processingLatency) {
+
+  override def chooseProcessor(subject: String): Option[JsValue => Future[Any]] =
+    PartialFunction.condOpt(subject) {
+      case "image-indexed" => processIndexedImage
+    }
+
+  def isArchivingCategory(category: String) = {
+    val archiveCategory = List("staff-photographer", "contract-photographer", "commissioned-photographer")
```

Comment:
You could match on the types.
```Scala
isArchivingUsageRights[T <: UsageRights](usageRights: T) = usageRights match {
  case _:StaffPhotographer | _:ContractPhotographer | _:CommissionedPhotographer => true
  case _ => false
}
```

(there might be a better syntax, but it's nicer than using the strings).